common/erc20-bridge-contract/src/keys.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+// Serializable structures for what we find in common/crypto
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, JsonSchema)]
+pub struct PublicKey([u8; 32]);
+
+impl PublicKey {
+    pub fn new(bytes: [u8; 32]) -> Self {
```

Comment:
I don't think this is the place to check for it though. This is just some dumb wrapper that helps sending bytes over the wire. There is a check that is done in the cosmos contract, in `link_payment`, when `ed25519_verify` is invoked. I changed a bit that error path now, so that it's more verbose.